pkg/kubectl/cmd/set/set_port.go

```diff
@@ -0,0 +1,281 @@
+		Example: portExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "identifying the resource to get from a server."
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+	cmd.Flags().StringVarP(&options.containerSelector, "containers", "c", "*", "The names of containers in the selected pod templates to change - may use wildcards")
+	cmd.Flags().BoolVar(&options.All, "all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set port will NOT contact api-server but run locally.")
+	cmd.Flags().Int("port", 80, i18n.T("Name or number for the port on the container that the service should direct traffic to. Optional."))
+	cmd.Flags().String("protocol", "tcp", i18n.T("The network protocol for the service to be created. Default is 'TCP'."))
+	cmd.Flags().BoolVar(&options.overwrite, "overwrite", true, "If true, allow environment to be overwritten, otherwise reject updates that overwrite existing environment.")
+	cmdutil.AddRecordFlag(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+	return cmd
+}
+
+func (o *PortOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.Mapper, o.Typer = f.Object()
+	o.UpdatePodSpecForObject = f.UpdatePodSpecForObject
+	o.portsForObject = f.PortsForObject
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = cmdutil.GetFlagString(cmd, "output") == "name"
+	o.Record = cmdutil.GetRecordFlag(cmd)
+	o.ChangeCause = f.Command(cmd, false)
+	o.PrintObject = f.PrintObject
+	o.Local = cmdutil.GetFlagBool(cmd, "local")
+	o.DryRun = cmdutil.GetDryRunFlag(cmd)
+	o.Output = cmdutil.GetFlagString(cmd, "output")
+	o.port = cmdutil.GetFlagInt(cmd, "port")
+	o.protocol = strings.ToUpper(cmdutil.GetFlagString(cmd, "protocol"))
+	o.containerSelector = cmdutil.GetFlagString(cmd, "containers")
+	o.overwrite = cmdutil.GetFlagBool(cmd, "overwrite")
+	o.Cmd = cmd
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	builder := f.NewBuilder(!o.Local).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		Flatten()
+	if !o.Local {
+		builder = builder.
+			SelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, args...).
+			Latest()
+	}
+	o.Infos, err = builder.Do().Infos()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *PortOptions) Validate() error {
+	errors := []error{}
+	if o.protocol != "TCP" && o.protocol != "UDP" {
+		errors = append(errors, fmt.Errorf("protocol must tcp or udp"))
+	}
+	if o.port < 0 {
```

Comment:
should be `o.port <=0`?

IIRC, we've got a range for such ports. Please find it in api validating.

Reply:
agree, i will follow server-side